src/wallet/wallet.cpp

```diff
@@ -1344,6 +1344,21 @@ void CWalletTx::GetAmounts(list<COutputEntry>& listReceived,
         nFee = nDebit - nValueOut;
     }
 
+    // Create an output for the value taken from the transparent value pool by JoinSplits
+    if (nDebit > 0) {
+        CAmount nValueOut = 0;
+        for (const JSDescription & js : vjoinsplit) {
+            nValueOut += js.vpub_old;
```

Comment:
Why do we only account for vpub_old and not vpub_new? Let's say we have a transaction with two JoinSplits, one with vpub_old of 2 ZEC and the other with vpub_new of 1 ZEC. That should be completely equivalent to having a single JoinSplit with vpub_old of 1 ZEC, right? It is an implementation detail of the JoinSplit construction algorithm which of those we choose (you can imagine the latter being an optimisation of/privacy improvement on the former). So this must be wrong because it distinguishes implementation details.